crypto/evp/keymgmt_lib.c

```diff
@@ -127,8 +128,8 @@ void *evp_keymgmt_util_export_to_provider(EVP_PKEY *pk, EVP_KEYMGMT *keymgmt)
      */
     if (pk->dirty_cnt == pk->dirty_cnt_copy) {
         /* If this key is already exported to |keymgmt|, no more to do */
-        op = evp_keymgmt_util_find_operation_cache(pk, keymgmt);
-        if (op != NULL && op->keymgmt != NULL) {
+        op = evp_keymgmt_util_find_operation_cache(pk, keymgmt, selection);
+        if (op != NULL && op->keymgmt != NULL && op->selection == selection) {
```

Comment:
why is this additional check needed?

Reply:
Just wanted to make sure, I can drop it if you think it is redundant.

Reply:
Yes please

Reply:
ok